appsody/builder.go

```diff
@@ -26,7 +26,7 @@ func NewBuilder(registry, namespace string) *Builder {
 }
 
 // Build an image from the function source at path.
-func (n *Builder) Build(name, path string) (image string, err error) {
+func (n *Builder) Build(name, language, path string) (image string, err error) {
```

Comment:
Deriving the language from the .faas.config didn't work out?

Reply:
I am not sure if this method should be aware of the config file. The config is read by caller of this function.

Reply:
That's fine.  We can just remember that the source code has a place for metadata like this if we want to lean towards introspection in the future.

Reply:
Maybe we could pass Function{} struct there?

Reply:
We could, but there are purposely zero dependencies from the parent into the implementation packages such that they are free to use exported utilities from the parent package.  So any implementation of a "Builder", for instance, can create a "Function" about a given path to get a typesafe interface to a meta-object representing that function, for instance.

Reply:
By structuring in this way, with the "Domain Types" in the root, but the root having no dependencies on the implementing packages, we can share certain abstractions like this between the subcomponents.  For instance, I could see us adding more and more metadata to the .faas.yaml as time goes on.  Any implementer of Builder, Updater, etc., by just creating a new `Function(path)` can then have access to this metadata.
I don't think passing language in as a parameter is necessarily wrong.  But it seems redundant since, in theory, the path points to a function whose language should be derivable.

Reply:
Isn't `name` redundant too then?

Reply:
That's a good point.  It's "Build the function at this path", not "Build the function at this path and use this name".
That was already set during the initialization phase when it was asked to "Initialize a function at this location with this name...".